### PR TITLE
Auth provider

### PR DESCRIPTION
**Pull request by bhujoshi**

### CODE REVIEW

**Comment by bhujoshi** [comment]
Basic UI build for login overlay working on functionality.
Screenshot 2024-04-15 at 10 20 21 AM
Screenshot 2024-04-15 at 10 20 10 AM

File: components/GridActionButton.android.js [hunk 1]
```diff
@@ -201,9 +201,10 @@ const GridActionButtonAndroid = ({ navigation, setPickImageLoading }) => {
           ]
         ]}></ActionButtonContainer>
       <Modal
+        style={{ height: 400 }}
```
Comment (kamalkishor1991) [inline components/GridActionButton.android.js:204]: We should make it percentage of screen height and also move it to styles

File: components/Login.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      res.data.user.full_name,
+      res.data.user.email,
+      res.data.userToken,
+      res.data.user.avatar_url
+    );
+    // user.loggedIn = true;
+    // setUser(user);
+    // naviagteAfterLogin();
+  } catch (error) {
+    // setError(error.message);
```
Comment (kamalkishor1991) [inline components/Login.js:28]: We should show errors like this to users also and send the error to server for us to debug 
https://github.com/croma-app/huehive-mobile-app/blob/7012a4971cf54586edf2ff2b33e05638081e0309/libs/Helpers.js#L20

File: screens/ChatSessionScreen.js [hunk 3]
```diff
@@ -339,4 +330,4 @@ const styles = StyleSheet.create({
   }
 });
 
-export default ChatSessionScreen;
+export default loginHoc(ChatSessionScreen);
```
Comment (kamalkishor1991) [inline screens/ChatSessionScreen.js:333]: nice.

**Review by kamalkishor1991 — CHANGES_REQUESTED** [review]

File: components/LoginWrapper.js [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+              )}
+            </View>
+          </View>
+        </View>
+      </TouchableWithoutFeedback>
+    </Modal>
+  );
+};
+
+const LoginWrapper = function ({ children }) {
```
Comment (kamalkishor1991) [inline components/LoginWrapper.js:114]: AppAuthProvider is a better name.

File: components/LoginWrapper.js [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+import { notifyMessage, sendClientError } from '../libs/Helpers';
+import { useTranslation } from 'react-i18next';
+import { googleLogin } from '../network/login-and-signup';
+
+const SCREEN_TYPES = {
+  LOGIN: 'LOGIN',
+  SIGN_UP: 'SIGN_UP'
+};
+
+const LoginOverlay = function () {
```
Comment (kamalkishor1991) [inline components/LoginWrapper.js:23]: AuthOverlay

File: components/LoginWrapper.js [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+      {children}
+    </>
+  );
+};
+
+const styles = StyleSheet.create({
+  login_overlay: {
+    position: 'absolute',
+    top: 0,
+    height: Dimensions.get('window').height,
```
Comment (kamalkishor1991) [inline components/LoginWrapper.js:129]: We should try to avoid depending on Dimensions in this case.

File: hooks/useLoginOverlay.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+import { create } from 'zustand';
+
+const useLoginOverlay = create((set) => ({
```
Comment (kamalkishor1991) [inline hooks/useLoginOverlay.js:3]: we can call it useAuth I think?

File: hooks/useUserData.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+const useUserData = create((set) => ({
+  userData: undefined,
+  isLoading: false,
+  loadUserData: async () => {
+    set({ isLoading: true });
+    try {
+      const userData = await retrieveUserSession();
+      set({ userData });
+    } catch (error) {
+      console.log('Error: Failed to load userData');
```
Comment (kamalkishor1991) [inline hooks/useUserData.js:13]: do we need catch here?

**Review by kamalkishor1991 — CHANGES_REQUESTED** [review]

File: components/ForgetPassword.js [hunk 9]
```diff
@@ -0,0 +1,93 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react';
```
[the remark this answers is not in the document]
Comment (bhujoshi) [reply components/ForgetPassword.js]: Removed

File: components/LoginWrapper.js (hunk 4 above)
Comment (bhujoshi) [reply components/LoginWrapper.js:114]: Changed

File: components/LoginWrapper.js (hunk 5 above)
Comment (bhujoshi) [reply components/LoginWrapper.js:23]: Changed

File: hooks/useLoginOverlay.js (hunk 7 above)
Comment (bhujoshi) [reply hooks/useLoginOverlay.js:3]: Done

File: hooks/useUserData.js (hunk 8 above)
Comment (bhujoshi) [reply hooks/useUserData.js:13]: removed

**Pull request merged**